c++/src/kj/async-win32.c++

```diff
@@ -133,13 +133,44 @@ private:
   Promise<IoResult> promise;
 };
 
+typedef LONG(NTAPI* NtSetInfoFuncPtr)(HANDLE, ULONG_PTR*, void*, ULONG, ULONG);
+
+NtSetInfoFuncPtr getSetInfoFileFunc()
+{
+    static thread_local NtSetInfoFuncPtr setInfoFileFunc = []() -> NtSetInfoFuncPtr {
```

Comment:
Why `thread_local`? Should be the same in all threads right? Note that static local initialization is thread-safe since C++11.

Reply:
Ah! Indeed, I did it because I feared it would not be thread-safe to use a non thread_local variable. Good to know that's not the case